src/otx/core/model/entity/detection.py

```diff
@@ -0,0 +1,138 @@
+        except AssertionError:
+            model = MODELS.build(convert_conf_to_mmconfig_dict(self.config, to="list"))
+
+        return model
+
+    def _customize_inputs(self, entity: DetBatchDataEntity) -> dict[str, Any]:
+        from mmdet.structures import DetDataSample
+        from mmengine.structures import InstanceData
+
+        mmdet_inputs: dict[str, Any] = {}
+
+        mmdet_inputs["inputs"] = entity.images  # B x C x H x W PyTorch tensor
+        mmdet_inputs["data_samples"] = [
+            DetDataSample(
+                metainfo={
+                    "img_id": img_info.img_idx,
+                    "img_shape": img_info.img_shape,
+                    "ori_shape": img_info.ori_shape,
+                    "pad_shape": img_info.pad_shape,
+                    "scale_factor": img_info.scale_factor,
+                },
+                gt_instances=InstanceData(
+                    bboxes=bboxes,
+                    labels=labels,
+                ),
+            )
+            for img_info, bboxes, labels in zip(
+                entity.imgs_info,
+                entity.bboxes,
+                entity.labels,
+            )
+        ]
+        preprocessor: DetDataPreprocessor = self.model.data_preprocessor
+        # Don't know why but data_preprocessor.device is not automatically
+        # converted by the pl.Trainer's instruction unless the model parameters.
+        # Therefore, we change it here in that case.
+        if preprocessor.device != (
+            model_device := next(self.model.parameters()).device
+        ):
+            preprocessor = preprocessor.to(device=model_device)
+            self.model.data_preprocessor = preprocessor
+
+        mmdet_inputs = preprocessor(data=mmdet_inputs, training=self.training)
+
+        mmdet_inputs["mode"] = "loss" if self.training else "predict"
+
+        return mmdet_inputs
+
+    def _customize_outputs(
+        self,
+        outputs: Any,  # noqa: ANN401
+    ) -> Union[DetBatchPredEntity, OTXBatchLossEntity]:
+        from mmdet.structures import DetDataSample
+
+        if self.training:
+            if not isinstance(outputs, dict):
+                raise TypeError(outputs)
+
+            losses = OTXBatchLossEntity()
+            for k, v in outputs.items():
+                losses[k] = sum(v)
+            return losses
+
+        scores = []
+        bboxes = []
+        labels = []
+
+        for output in outputs:
+            if not isinstance(output, DetDataSample):
+                raise TypeError(output)
+
+            scores.append(output.pred_instances.scores)
+            bboxes.append(
+                tv_tensors.BoundingBoxes(
+                    output.pred_instances.bboxes,
+                    format="XYXY",
+                    canvas_size=output.img_shape,
+                ),
+            )
+            labels.append(output.pred_instances.labels)
+
+        return DetBatchPredEntity(
+            batch_size=len(outputs),
+            images=[],
```

Comment:
Why are `images` and `imgs_info` initialized as empty lists in `DetBatchPredEntity`? The following use case appears to be more intuitive:

```python
data_sample = DetDataSample()
instance_data = InstanceData()
instance_data.scores = scores
instance_data.bboxes = bboxes
instance_data.labels = labels
data_sample.pred_instances = instance_data
```

Reply:
Yea, we can pass the images also to support the visualization.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/2670/commits/c2dc6748799f14d5fc4d681e6b6992246c2b9b43 will not let them empty.